test/inductor/test_cpu_repro.py

```diff
@@ -4132,14 +4132,16 @@ def forward(self, x):
                 expected = mod(x)
                 compiled_m = torch.compile(mod)
                 actual, code = run_and_get_cpp_code(compiled_m, x)
-                self.assertEqual(expected, actual, atol=1e-4, rtol=1e-5)
+                self.assertEqual(expected, actual)
                 # 3 generated kernels (first one for var_mean, last two for result)
                 check_metrics_vec_kernel_count(3)
                 # check that there is no outer loop fusion.
                 self.assertEqual(
                     len(metrics.cpp_outer_loop_fused_inner_counts),
                     0,
                 )
+                # check for parallel reduction.
+                FileCheck().check("tmp_acc0_vec_arr").run(code)
```

Comment:
Suggest to add some metrics to check the parallel reduction codegen instead of checking the string patterns which are implementation specific. This eases the maintenance.

Reply:
Added, thanks for your comments!